sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java

```diff
@@ -276,28 +300,47 @@ public SpringAppDeploymentImpl withInstance(int count) {
     @Override
     public SpringAppDeploymentImpl withCpu(int cpuCount) {
         ensureDeploySettings();
-        innerModel().properties().deploymentSettings().withCpu(cpuCount);
+        if (innerModel().properties().deploymentSettings().resourceRequests() == null) {
+            innerModel().properties().deploymentSettings().withResourceRequests(new ResourceRequests());
+        }
+        innerModel().properties().deploymentSettings().resourceRequests().withCpu(String.valueOf(cpuCount));
         return this;
     }
 
     @Override
     public SpringAppDeploymentImpl withMemory(int sizeInGB) {
         ensureDeploySettings();
-        innerModel().properties().deploymentSettings().withMemoryInGB(sizeInGB);
+        if (innerModel().properties().deploymentSettings().resourceRequests() == null) {
+            innerModel().properties().deploymentSettings().withResourceRequests(new ResourceRequests());
+        }
+        innerModel().properties().deploymentSettings().resourceRequests().withMemory(String.format("%dGi", sizeInGB));
         return this;
     }
```

Comment:
Do we plan a String overload in later PR?

Reply:
Yes, will do in later PR.